qiskit_experiments/data_processing/data_processor.py

```diff
@@ -69,48 +103,79 @@ def is_trained(self) -> bool:
 
         return True
 
-    def __call__(self, data: Union[Dict, List[Dict]], **options) -> Tuple[Any, Any]:
+    def __call__(self, data: Union[Dict, List[Dict]], **options) -> Tuple[np.ndarray, np.ndarray]:
         """
         Call self on the given datum. This method sequentially calls the stored data actions
         on the datum.
 
         Args:
-            data: The data, typically from ExperimentData.data(...), that needs to be processed.
-            This dict or list of dicts also contains the metadata of each experiment.
+            data: The data, typically from ``ExperimentData.data(...)``,
+                that needs to be processed. This dict or list of dicts also contains
+                the metadata of each experiment.
             options: Run-time options given as keyword arguments that will be passed to the nodes.
 
         Returns:
             processed data: The data processed by the data processor.
         """
-        return self._call_internal(data, **options)
+        # Convert into conventional data format, This is temporary code block.
+        # Once following steps support ufloat array, data will be returned as-is.
+        # TODO need upgrade of following steps, i.e. curve fitter
+        processed_data = self._call_internal(data, **options)
+
+        try:
+            nominals = unp.nominal_values(processed_data)
+            stdevs = unp.std_devs(processed_data)
+            if np.isnan(stdevs).all():
+                stdevs = None
+        except TypeError:
+            # unprocessed data, can be count dict array.
+            nominals = processed_data
+            stdevs = None
```

Comment:
This looks okay. I'm wondering if `isinstance` would be better?

Reply:
I don't this it matters since this will be removed shortly.